Core/Core/Common/CommonModels/AppEnvironment/AppEnvironmentOverride.swift

```diff
@@ -127,6 +127,26 @@ extension AppEnvironment {
 
 // MARK: - Utils
 
+public struct RoutePath {
+    let value: String
+    let env: AppEnvironment
+
+    /// This method returns the correct path using env `baseURL`
+    /// if it was different from the one used on `AppEnvironment.shared.api`
+    func correctedPath() -> String {
+        guard env.api.baseURL != AppEnvironment.shared.api.baseURL else {
+            return value
+        }
+        return env.api.baseURL.appending(path: value).absoluteString
```

Comment:
Since we are working with strings can we add some logic if the input is already a fully qualified url and not just a path? Passing a full url here will definitely screw things up. Also, we could reinforce this solution with some unit tests.